tests/sqlAst/data/sql-ast-narrowing.php

```diff
@@ -214,28 +214,61 @@ public function strcase(PDO $pdo): void
         assertType('PDOStatement<array{col: non-empty-string&numeric-string, 0: non-empty-string&numeric-string}>', $stmt);
     }
 
-    public function avgMinMax(PDO $pdo): void
+    public function avg(PDO $pdo): void
     {
         $stmt = $pdo->query('SELECT avg(freigabe1u1) as avg from ada');
-        assertType('PDOStatement<array{avg: int<-32768, 32767>|null, 0: int<-32768, 32767>|null}>', $stmt);
+        assertType('PDOStatement<array{avg: numeric-string|null, 0: numeric-string|null}>', $stmt);
 
+        $stmt = $pdo->query('SELECT avg(eladaid) as avg from ak');
+        assertType('PDOStatement<array{avg: numeric-string|null, 0: numeric-string|null}>', $stmt);
+
+        $stmt = $pdo->query('SELECT avg(coalesce(eladaid, 9999999999999999)) as avg from ak');
+        assertType('PDOStatement<array{avg: numeric-string|null, 0: numeric-string|null}>', $stmt);
+
+        $stmt = $pdo->query('SELECT avg(email) as avg from ada');
+        assertType('PDOStatement<array{avg: float|null, 0: float|null}>', $stmt);
+
+        // Test non-empty-string input
+        $stmt = $pdo->query('SELECT avg(concat("0")) as avg from ada');
```

Comment:
isn't `concat("0") a `numeric-string`?

Reply:
It's both a `numeric-string` and `non-empty-string`:
```
string(39) "SELECT avg(concat("0")) as avg from ada"
object(PHPStan\Type\IntersectionType)#914 (2) {
  ["sortedTypes":"PHPStan\Type\IntersectionType":private]=>
  bool(false)
  ["types":"PHPStan\Type\IntersectionType":private]=>
  array(3) {
    [0]=>
    object(PHPStan\Type\StringType)#896 (0) {
    }
    [1]=>
    object(PHPStan\Type\Accessory\AccessoryNumericStringType)#897 (0) {
    }
    [2]=>
    object(PHPStan\Type\Accessory\AccessoryNonEmptyStringType)#891 (0) {
    }
  }
}
```

Reply:
when you use `$type->describe(VerbosityLevel::precise())` it will only print `numeric-string` as all `numeric-string`s are non empty.

I think its impossible to create a non-empty-string with a constant value. you either get a numeric-string or a non-falsey-string :)

lets just fix the comment